crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -241,12 +232,13 @@ pub struct P2PStream<S> {
     /// The supported capability for this stream.
     shared_capabilities: SharedCapabilities,
 
-    /// Outgoing messages buffered for sending to the underlying stream.
-    outgoing_messages: VecDeque<Bytes>,
+    /// A ping message that needs to be sent but couldn't be sent yet because the sink
+    /// was not ready.
+    pending_ping: Option<Bytes>,
 
-    /// Maximum number of messages that we can buffer here before the [Sink] impl returns
-    /// [`Poll::Pending`].
-    outgoing_message_buffer_capacity: usize,
+    /// A disconnect message that needs to be sent but couldn't be sent yet because the sink
+    /// was not ready.
+    pending_disconnect: Option<Bytes>,
```

Comment:
Btw before merging, I realized we could maybe have only one `pending_message`, because `pending_disconnect` will be `Some` only after clearing the ping message anyway (see line 312)